### PR TITLE
feat: orchestrator switches to FDv1 fallback on directive

### PR DESCRIPTION
**Pull request by beekld**
## Summary

Implements the FDv1 fallback directive in the FDv2 data system with TTL-based recovery.

- `FDv2SourceResult` carries `std::optional<FDv1FallbackDirective>` with a TTL (default 1h, 0 = indefinite).
- Sources parse `X-LD-FD-Fallback-TTL` (header) and `protocolFallbackTTL` (goodbye).
- On directive: switch to FDv1 if configured, otherwise disconnect (`kInterrupted`). Schedule retry after TTL.
- On TTL elapse: `SourceManager::SwitchBackToFDv2()` re-enables FDv2 (including previously terminal-error-blocked factories), blocks FDv1, restarts synchronizers.

End-to-end fallback will work once an FDv1 streaming source is wrapped as an `IFDv2Synchronizer` (#540).

## Test plan

- [x] `SourceManager` tests cover the new block/unblock semantics
- [x] Source-level tests cover TTL parsing (HTTP + goodbye), default, malformed
- [x] Orchestrator tests cover directive flows in initializer/synchronizer paths, TTL elapse rebuild, FDv2 recovery
- [x] Full server suite green

---

> [!NOTE]
> **Medium Risk**
> Changes core flag-data orchestration and protocol fallback behavior; mis-handling could leave clients on the wrong protocol or retry timing, though behavior is heavily tested.
> 
> **Overview**
> Adds **server-directed FDv1 fallback** for the FDv2 data path: results now carry an optional `FDv1FallbackDirective` (TTL, default 1h; **0 = stay on FDv1 indefinitely**) instead of a boolean flag.
> 
> **Parsing:** Goodbye JSON gains `protocolFallbackTTL`; polling and streaming read `X-LD-FD-Fallback` / `X-LD-FD-Fallback-TTL`, with body/goodbye directives overriding headers.
> 
> **Orchestration:** On directive, `SourceManager` blocks FDv2 factories and selects an `IsFDv1Fallback()` synchronizer; missing adapter → `kInterrupted` without forcing `kOff`. A cancellable timer calls `SwitchBackToFDv2()` (re-enabling FDv2 factories blocked for terminal errors) and restarts synchronizers. Destructor/`Close()` no longer sets status to `kOff`; retry timers are cancelled on close.
> 
> **Other:** Logging tweaks (initializer/synchronizer start, deduped synchronizer interrupts). Broad unit/orchestrator tests cover TTL parsing, switching, recovery, and edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 81b1c7fa7f95c2f1365680b5d48f3d07ae8025c8. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: libs/server-sdk/src/data_systems/fdv2/fdv2_data_system.cpp [hunk 1]
```diff
@@ -185,8 +193,30 @@ void FDv2DataSystem::OnInitializerResult(
         if (closed_ || got_shutdown) {
             return;
         }
+        if (result.fdv1_fallback) {
+            source_manager_.SwitchToFDv1Fallback();
+            if (source_manager_.AvailableSynchronizerCount() > 0) {
+                LD_LOG(logger_, LogLevel::kInfo)
+                    << Identity() << ": FDv1 fallback engaged";
+                got_basis = true;
```
Comment (kinyoklion) [inline libs/server-sdk/src/data_systems/fdv2/fdv2_data_system.cpp:204]: Feels worthy of a comment potentially.

File: libs/server-sdk/src/data_systems/fdv2/streaming_synchronizer.cpp [hunk 2]
```diff
@@ -240,7 +253,14 @@ void FDv2StreamingSynchronizer::State::OnEvent(sse::Event const& event) {
...
                     << r.reason.value_or("") << "'.";
-                Notify(FDv2SourceResult{FDv2SourceResult::Goodbye{r.reason}});
+                FDv2SourceResult goodbye_result{
+                    FDv2SourceResult::Goodbye{r.reason}};
+                if (r.protocol_fallback_ttl) {
+                    goodbye_result.fdv1_fallback =
+                        data_interfaces::FDv1FallbackDirective{
+                            std::chrono::seconds(*r.protocol_fallback_ttl)};
+                }
+                Notify(std::move(goodbye_result));
```
[the remark this answers is not in the document]
Comment (kinyoklion) [reply libs/server-sdk/src/data_systems/fdv2/streaming_synchronizer.cpp:263]: Seems minor.

**Review by kinyoklion — APPROVED** [review]

File: libs/server-sdk/src/data_systems/fdv2/fdv2_data_system.cpp (hunk 1 above)
Comment (beekld) [reply libs/server-sdk/src/data_systems/fdv2/fdv2_data_system.cpp:204]: added a comment

File: libs/server-sdk/src/data_systems/fdv2/fdv2_data_system.cpp [hunk 3]
```diff
@@ -349,20 +391,81 @@ void FDv2DataSystem::OnSynchronizerResult(
...
+            [this](bool fired) -> std::monostate {
+                if (fired) {
+                    OnFDv1RetryTimer();
+                }
+                return {};
+            },
+            [ioc = ioc_](async::Continuation<void()> work) {
+                boost::asio::post(ioc, std::move(work));
+            });
+}
```
[the remark this answers is not in the document]
Comment (beekld) [reply libs/server-sdk/src/data_systems/fdv2/fdv2_data_system.cpp:454]: I don't think that can actually happen, but I added a defensive check just in case it does.
